lambda/recipes-responder/src/takedown_processor.ts

```diff
@@ -1,29 +1,22 @@
 import type {DeletedContent} from "@guardian/content-api-models/crier/event/v1/deletedContent";
 import type {Event} from "@guardian/content-api-models/crier/event/v1/event"
-import {ContentType} from "@guardian/content-api-models/v1/contentType";
-import {awaitableDelay, removeAllRecipesForArticle} from "@recipes-api/lib/recipes-data";
+import {removeAllRecipesForArticle} from "@recipes-api/lib/recipes-data";
 import {DynamoClient} from "./dynamo_conn";
+import {ItemType} from "@guardian/content-api-models/crier/event/v1/itemType";
 
-export async function handleTakedown(evt:Event, attempt?:number):Promise<void> {
-  if(evt.payload?.kind=="content") {  //we don't need to handle if it's not an article
-    const content = evt.payload.content;
-    if(content.type==ContentType.ARTICLE) {  //we don't need to handle if it's not an article
-      try {
-        await removeAllRecipesForArticle(DynamoClient, content.id);
-      } catch(err) {
-        if( (attempt ?? 0) > 10) {
-          throw new Error(`Could not remove recipes for taken-down article ${content.id} after 10 attempts, giving up`);
-        }
-        console.error(`WARNING: Could not remove recipes for taken down article ${content.id}: `, err);
-        await awaitableDelay();
-        return handleTakedown(evt, (attempt ?? 0)+1)
-      }
-    }
+export async function handleTakedown(evt:Event):Promise<number> {
+  console.log("takedown payload: ", JSON.stringify(evt));
+
+  if(evt.itemType==ItemType.CONTENT) {
+    //there's no payload in the takedown message!
+    return removeAllRecipesForArticle(DynamoClient, evt.payloadId); //evt.payloadId is the canonical article ref that was taken down
+  } else {
+    return 0;
   }
 }
 
-// eslint-disable-next-line @typescript-eslint/require-await -- not implemented yet
-export async function handleDeletedContent(evt:DeletedContent):Promise<void> {
-
-  throw new Error("handleTakedown is not implemented yet")
+// I don't think that these are relevant to us here. So, I'm logging it out to verify that suspicion
+export async function handleDeletedContent(evt:DeletedContent):Promise<number> {
```

Comment:
Deletes are much rarer than takedowns but I believe they can still happen, so this logging may not catch one in the time we give it. In addition to the log action, we could also make a call from here to `handleTakedown` and treat it the same way, at least until we've proved whether deletes still flow out of Composer like they used to.

Reply:
The interesting thing here is that the only thing we get is the `DeletedContent` event which contains `aliasPaths` but no canonical paths. I'm therefore not sure _what_ should get deleted in this context (because we are only interested in the canonical path rather than any alias paths).